Tools/ADT/ci_shard_report.py

```
@@ -0,0 +1,58 @@
+def main(root):
+    paths = sorted(glob.glob(os.path.join(root, "**", "*.xml"), recursive=True))
+    if not paths:
+        print(f"No NUnit XML found under {root}")
+        return 1
+
+    seen = defaultdict(list)
+    rows = []
+
+    for path in paths:
+        shard = shard_name(path)
+        run = ET.parse(path).getroot()
```

Ruff (0.16.1)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)